torchvision/prototype/transforms/_augment.py

```diff
@@ -380,7 +409,95 @@ def forward(self, *inputs: Any) -> Any:
+
+    def _check_inputs(self, flat_inputs: List[Any]) -> None:
+        if has_any(flat_inputs, features.Mask, features.Video):
+            raise TypeError(f"{type(self).__name__}() is only supported for images and bounding boxes.")
+
+        if not (
+            has_any(flat_inputs, features.Image, PIL.Image.Image, features.is_simple_tensor)
+            and has_any(flat_inputs, features.BoundingBox)
+        ):
+            raise TypeError(f"{type(self).__name__}() is only defined for tensor images and bounding boxes.")
+
+    def forward(self, *inputs: Any) -> Any:
+        flat_batch_with_spec, images, targets = flatten_and_extract_data(
+            inputs,
+            boxes=(features.BoundingBox,),
+            labels=(features.Label, features.OneHotLabel),
+        )
+        # TODO: refactor this since we have already extracted the images and boxes
+        self._check_inputs(flat_batch_with_spec[0])
+
+        # images = [t1, t2, ..., tN]
+        # Let's define paste_images as shifted list of input images
+        # paste_images = [tN, t1, ..., tN-1,]
+        images_rolled = images[-1:] + images[:-1]
+        targets_rolled = targets[-1:] + targets[:-1]
+
+        output_images, output_targets = [], []
+        for image_1, target_1, image_2, target_2 in zip(images, targets, images_rolled, targets_rolled):
+            output_image, output_target = self._mixup(
+                image_1,
+                target_1,
+                image_2,
+                target_2,
+            )
+            output_images.append(output_image)
+            output_targets.append(output_target)
+
+        return unflatten_and_insert_data(flat_batch_with_spec, output_images, output_targets)
+
+    def _mixup(
+        self,
+        image_1: features.TensorImageType,
+        target_1: Dict[str, Any],
+        image_2: features.TensorImageType,
+        target_2: Dict[str, Any],
+    ) -> Tuple[features.TensorImageType, Dict[str, Any]]:
+        """
+        Performs mixup on the given images and targets.
+        """
+        mixup_ratio = self._dist.sample().item()
+        print(mixup_ratio)
+
+        c_1, h_1, w_1 = image_1.shape
+        c_2, h_2, w_2 = image_2.shape
+        h_mixup = max(h_1, h_2)
+        w_mixup = max(w_1, w_2)
+
+        if mixup_ratio >= 1.0:
+            return image_1, target_1
+
+        # mixup images and prevent the object aspect ratio from changing
+        mix_img = torch.zeros(c_1, h_mixup, w_mixup, dtype=torch.float32)
+        mix_img[:, : image_1.shape[1], : image_1.shape[2]] = image_1 * mixup_ratio
+        mix_img[:, : image_2.shape[1], : image_2.shape[2]] += image_2 * (1.0 - mixup_ratio)
+        # mixup targets
+        mix_target = {**target_1, **target_2}
+        box_format = target_1["boxes"].format
+        mixed_boxes = {
+            "boxes": features.BoundingBox(
+                torch.vstack((target_1["boxes"], target_2["boxes"])),
```

Comment:
Why do we need to `vstack` here? This creates an extra dimension, doesn't it? Shouldn't it be

```suggestion
                torch.cat((target_1["boxes"], target_2["boxes"]), dim=-1),
```

like below for the labels?